csm/core/services/s3/server_info.py

```diff
@@ -15,18 +15,26 @@
 
 from csm.common.services import ApplicationService
 from csm.common.service_urls import ServiceUrls
+from csm.common.errors import InvalidRequest
 
 
 class S3ServerInfoService(ApplicationService):
     """
     Service for acessing S3 server information
     """
 
-    async def get_s3_server_info(self):
+    async def get_s3_server_info(self, schemas):
         """
         Obtain information about S3 server in json format
         """
-        s3_urls = ServiceUrls.get_s3_uris()
+        supported_schemas = ServiceUrls.get_s3_supported_schemas()
+        if schemas is not None:
+            if not all(s in supported_schemas for s in schemas):
+                raise InvalidRequest(f'Unsupported schema in list of schemas {schemas}. '
+                    f'Supported schemas are {supported_schemas}')
+            s3_urls = [ServiceUrls.get_s3_uri(s) for s in schemas]
+        else:
+            s3_urls = [ServiceUrls.get_s3_uri(s) for s in supported_schemas]
```

Comment:
I have just realized that a minor rewrite here to get rid of the very similar `s3_urls = [ServiceUrls.get_s3_uri(s) ...` lines and move them out of the conditional is possible.  This is not a required change, though.